qiskit/circuit/quantumcircuit.py

```diff
@@ -832,6 +833,11 @@ def tensor(self, other):
         dest.compose(other, range(other.num_qubits), range(other.num_clbits), inplace=True)
         dest.compose(self, range(other.num_qubits, num_qubits),
                      range(other.num_clbits, num_clbits), inplace=True)
+
+        # Replace information from tensored circuit into self when inplace = True
+        if inplace:
+            self.__dict__.update(dest.__dict__)
+            return None
```

Comment:
This will work in implementing the `inplace=True` interface (`.compose` was initially implemented the same way), but under the hood will still make a copy of `self` which would be better to avoid. This doesn't have to be added here, but should be taken care of as part of #6068 .